File: app/src/main/java/uk/co/amlcurran/connascence/Converter.java
package uk.co.amlcurran.connascence;

import java.text.DecimalFormat;

public class Converter {

    private static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat("#.00");
    private static Converter instance;

    public static Converter get() {
        if (instance == null) {
            instance = new Converter();
        }
        return instance;
    }

    public double poundsToEuros() {
        return 1.1;
    }

    public double eurosToPounds() {
        return 1 / poundsToEuros();
    }

    double convertPenceToEuros(int pence) {
        double pounds = pence / 100d;
        return Double.valueOf(DECIMAL_FORMAT.format(pounds * poundsToEuros()));
    }

    int convertEurosToPence(int euros) {
        double pounds = euros * eurosToPounds();
        return (int) (pounds * 100);
    }
}

File: ios/Connascence/Connascence/Converter.swift
import UIKit

class Converter {
    
    static let sharedInstance = Converter()
    
    func poundsToEuros(pounds: Double) -> Double {
        return pounds * 1.1
    }
    
    func eurosToPounds(euros: Double) -> Double {
        return euros / 1.1
    }

}
